src/D2L.Security.OAuth2/Utilities/D2LHttpClient.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace D2L.Security.OAuth2.Utilities
+{
+	public class D2LHttpClient : ID2LHttpClient
+	{
+		/// <summary>
+		/// This class implements sync versions of HTTPClient methods.
+		/// TODO: Replace the sync methods which are currently using Task.Run(async verison); task.Wait();
+		/// With (probably) HttpWebRequest
+		/// </summary>
+		private readonly HttpClient m_httpClient;
+
+		public D2LHttpClient() {
+			m_httpClient = new HttpClient();
+		}
+		public D2LHttpClient(HttpMessageHandler handler) {
+			m_httpClient = new HttpClient(handler);
+		}
+		public D2LHttpClient(HttpMessageHandler handler, bool disposeHandler) { 
+			m_httpClient = new HttpClient(handler, disposeHandler);
+		}

```

Comment:
For the moment I would suggest making `ID2LHttpClient` and `D2LHttpClient` `internal`, using it to wrap the passed in `HttpClient`, and avoid changing the public interface of this library (as far as the factories go and whatnot (i.e. undo changes to `AccessTokenProviderFactory`)).

```suggestion
		public D2LHttpClient( HttpClient httpClient ) {
			m_httpClient = httpClient;
		}
```